qiskit/pulse/instruction_schedule_map.py

```diff
@@ -105,7 +108,7 @@ def qubit_instructions(self, qubits: Union[int, Iterable[int]]) -> List[str]:
             return list(self._qubit_instructions[_to_tuple(qubits)])
         return []
 
-    def has(self, instruction: str, qubits: Union[int, Iterable[int]]) -> bool:
+    def has(self, instruction: Union[str, Gate], qubits: Union[int, Iterable[int]]) -> bool:
```

Comment:
Should this be a `Gate` or a [Instruction](https://github.com/Qiskit/qiskit-terra/blob/8c40b0250a35293ea923cddc23949c1d8628f512/qiskit/circuit/instruction.py#L49). Same in other places.

Reply:
This is a question I wanted to discuss with you. Because instructions did not have `_to_matrix` functionality I only used gates for my project, but since they inherit `instruction` should I just use instructions everywhere instead of `Gate`? Is there any reason to limit it to just `Gate`

Reply:
I think so because we often input `measure` which is an instruction.

Reply:
switched to instructions. I am still leaving the as `test_has_gate` and so on for each of the normal tests, since I test with a gate, but I assume that's fine because gates are a type of instruction that is easily accessible?

Reply:
Looks good.